wemake_python_styleguide/visitors/ast/conditions.py

```diff
@@ -188,3 +190,34 @@ def visit_Compare(self, node: ast.Compare) -> None:
             self.add_violation(refactoring.ChainedIsViolation(node))
 
         self.generic_visit(node)
+
+
+@final
+class SimplifiableMatchVisitor(BaseNodeVisitor):
+    """Checks for match statements that can be simplified to if/else."""
+
+    def visit_Match(self, node: ast.Match) -> None:
+        """Checks match statements."""
+        self._check_simplifiable_match(node)
+        self.generic_visit(node)
+
+    def _check_simplifiable_match(self, node: ast.Match) -> None:
+        cases = node.cases
+        if len(cases) == 2:
+            first, second = cases
+
+            if (
+                pattern_matching.is_wildcard_pattern(second)
+                and first.guard is None
+                and pattern_matching.is_simple_pattern(first.pattern)
+            ):
+                self.add_violation(consistency.SimplifiableMatchViolation(node))
+                return
+
+            if (
+                pattern_matching.is_wildcard_pattern(second)
+                and first.guard is not None
```

Comment:
```suggestion
```

I think that we can drop any `guard` references. When patterns are simple, we can simplify them with and without guards. When patterns are complex - we also don't care about guards.

So, the whole code can be:

```python
if not pattern_matching.is_wildcard_pattern(second):
    return

if (
    pattern_matching.is_irrefutable_binding(first.pattern)
    or pattern_matching.is_simple_pattern(first.pattern)
):
    self.add_violation(consistency.SimplifiableMatchViolation(node))
```